tests/yakus/concealedpon.test.cpp

```diff
@@ -0,0 +1,90 @@
+#include <gtest/gtest.h>
+#include "hands.h"
+#include "hand.h"
+#include "gamestate.h"
+#include "handformer.h"
+#include "handnode.h"
+#include "analysis.h"
+#include "statefunctions.h"
+
+using Mahjong::GameState;
+using Mahjong::Hand;
+using Mahjong::HandFromNotation;
+using Mahjong::breakdownHand;
+using Mahjong::Node;
+using Mahjong::Piece;
+using Mahjong::Meld;
+
+TEST(isThreeConcealedPons, 2Han) {
+  auto gameState = GameState();
+  gameState.hands[0] = Hand(HandFromNotation("111m111p111s666z44m"));
+  gameState.hands[0].open = false;
+
+  auto root = breakdownHand(gameState.hands.at(0).live);
+
+  for (const auto& branch : Node::AsBranchVectors(root.get())) {
+    if (Mahjong::isThreeConcealedPons(gameState, 0, branch) == 2) {
+      SUCCEED();
+      return;
+    }
+  }
+  FAIL();
+}
+
+TEST(isThreeConcealedPons, PonsMustBeConcealed) {
```

Comment:
You should add a test checking that 3 concealed pons with an additional melded pon still counts for the yaku.

Reply:
Added in 8ca3870ae0bf49165c9fc1220ded49941f8f0ccc